aquarius/events/events_monitor.py

```diff
@@ -181,51 +182,60 @@ def process_block_range(self, from_block, to_block):
             self._chain_id,
         ]
 
-        for event in self.get_event_logs(EVENT_METADATA_CREATED, from_block, to_block):
-            dt_contract = self._web3.eth.contract(
-                abi=ERC721Template.abi, address=event.address
-            )
-            receipt = self._web3.eth.get_transaction_receipt(
-                event.transactionHash.hex()
-            )
-            event_object = dt_contract.events.MetadataCreated().processReceipt(receipt)[
-                0
-            ]
-            try:
-                event_processor = MetadataCreatedProcessor(
-                    *([event_object, dt_contract, receipt["from"]] + processor_args)
-                )
-                event_processor.process()
-            except Exception as e:
-                logger.exception(
-                    f"Error processing create metadata event: {e}\n" f"event={event}"
-                )
+        event_processors = {
+            "EVENT_METADATA_CREATED": MetadataCreatedProcessor,
+            "EVENT_METADATA_UPDATED": MetadataUpdatedProcessor,
+            "EVENT_METADATA_STATE": MetadataStateProcessor,
+        }
 
-        for event in self.get_event_logs(EVENT_METADATA_UPDATED, from_block, to_block):
-            dt_contract = self._web3.eth.contract(
-                abi=ERC721Template.abi, address=event.address
-            )
-            receipt = self._web3.eth.get_transaction_receipt(
-                event.transactionHash.hex()
-            )
-            event_object = dt_contract.events.MetadataUpdated().processReceipt(receipt)[
-                0
-            ]
-            try:
-                event_processor = MetadataUpdatedProcessor(
-                    *([event_object, dt_contract, receipt["from"]] + processor_args)
-                )
-                event_processor.process()
-            except Exception as e:
-                logger.error(
-                    f"Error processing update metadata event: {e}\n" f"event={event}"
-                )
+        self.handle_events_processors(
+            event_processors, processor_args, from_block, to_block
+        )
 
         self.handle_order_started(from_block, to_block)
+
         self.store_last_processed_block(to_block)
 
+    def handle_events_processors(
+        self, event_processors_mapping, processor_args, from_block, to_block
+    ):
+        """Process emitted events between two given blocks from a mapping of events
+        and processors.
+
+        Args:
+            event_processors_mapping (Dict[str, EventProcessor]): event names and associated processors
+            processor_args (List[any]): list of processors arguments
+            from_block (int): inital block
+            to_block (int): final block
+        """
+        for event_name in event_processors_mapping:
```

Comment:
I would move the outer for outside the function and call this function handle_regular_event_processor. That way, there is consistency between the handle_order_started and handle_regular_event_processor (they all treat one type of event), instead of bulking three types of events in one function. It will also help with readability:
- no need to pass event_processors_mapping as a parameter
- one level of indentation removed